modules/matterhorn-workflow-service-impl/src/main/java/org/opencastproject/workflow/impl/WorkflowServiceImpl.java

```diff
@@ -1032,6 +1032,10 @@ private void removeTempFiles(WorkflowInstance workflowInstance) throws WorkflowD
           UnauthorizedException, NotFoundException {
     logger.info("Removing temporary files for workflow %s", workflowInstance);
     for (MediaPackageElement elem : workflowInstance.getMediaPackage().getElements()) {
+      if (null == elem.getURI()) {
+        // should never pass here but just in case
```

Comment:
@wsmirnow Maybe log a warning?

The problem with this kind of code is that it masks errors that happened elsewhere: At the end of the day, the media package is in an invalid state if we reached this code, isn't it? While it makes sense to "continue deletion", we still might want to indicate that there is actually something wrong here...  even if just to remind us that another bug somewhere out there is causing an inconsistency in our media package.

What do you think?

Reply:
Hi @staubesv,
I'm with you. But I wouldn't write this patch if this case was never happened…

What do you thik about a warning message like:
Mediapackage element {} from media package {} does not have an URI set. There might be a bug in the Opencast code.

The second sentence is not helpfull for users and might be removed.

Reply:
Please omit the "There might be a bug ..." in the message. The rest seems fine.